code-challenges.test.js

```diff
@@ -13,55 +13,131 @@
+  })
+})
 
-const studyMinutesWeek1 = {
-  sunday: 90,
-  monday: 30,
-  tuesday: 20,
-  wednesday: 15,
-  thursday: 30,
-  friday: 15,
-  saturday: 60
-}
-// Expected output: [15, 15, 20, 30, 30, 60, 90]
-
-const studyMinutesWeek2 = {
-  sunday: 100,
-  monday: 10,
-  tuesday: 45,
-  wednesday: 60,
-  thursday: 20,
-  friday: 15,
-  saturday: 65
-}
-// Expected output: [10, 15, 20, 45, 60, 65, 100]
 
 // b) Create the function that makes the test pass.
+//create a function named leastGreatest that takes an object as a parameter
+const leastGreatest = (object) => {
+  //create a variable named sortedNumbers that hold the value of the 
+  //objects value using Object.values()
+  //sort the numbers in ascending order using the .sort method
+  //within the .sort() method we need to distinguish that we want
+  //the numbers in asending order so we assign variables a and b
+  //we place a before b to get the ascending values.
+  let sortedNumbers = Object.values(object).sort((a,b)=>a-b)
+  //return all of the sorted numbers
+  return sortedNumbers
+}
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
+describe('accSum', () => {
+  it('takes in an array and returns an array of the accumulating sum. An empty array should return an empty array', () => {
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+    const accountTransactions1 = [100, -17, -23, -9]
+    // Expected output: [100, 83, 60, 51]
+    const accountTransactions2 = [250, -89, 100, -96]
+    // Expected output: [250, 161, 261, 165]
+    const accountTransactions3 = []
+    // Expected output: []
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+    expect(accSum(accountTransactions1)).toEqual([100, 83, 60, 51])
+    expect(accSum(accountTransactions2)).toEqual([250, 161, 261, 165])
+    expect(accSum(accountTransactions3)).toEqual([])
+  })
+})
 
-const accountTransactions3 = []
-// Expected output: []
 
 // b) Create the function that makes the test pass.
+//create a function named accSum that takes an array as a parameter
+const accSum = (array) => {
+  //create a variable named sum that will be used for for holding sum value
+  let sum = 0
+  //if the array is empty return an empty string
+  if(array.length===0){
+    return []
+    //if the array is not empty use .map() method to iterate through the 
+    //array. the sum variable starts at 0 and recieves a new value after
+    //adding/subtracting the value, that number is placed into a separate
+    //array. once it is done we return that separate array with values that
+    // have been accumulatively summated ---> hope those are real words I want to sound fancy
+  } else if (array.length>0){
+    return array.map((value) => sum += value)
```

Comment:
I don’t know if those are real words but this is great! Love the approach👏🏽 Nice use of the addition assignment operator